perses/rjmc/geometry.py

```diff
@@ -525,34 +560,37 @@ def _propose_bond(self, bond, beta):
         """
         Bond length proposal
         """
-        k_eq = bond.type.k*units.kilocalories_per_mole/(units.angstrom**2)
-        r0 = bond.type.req*units.angstrom
-        sigma = beta*2.0/np.sqrt(2.0*k_eq/k_eq.unit)
-        r = sigma/sigma.unit*np.random.random()*r0.unit + r0
+        r0 = bond.type.req
+        k = bond.type.k
+        sigma_r = units.sqrt(1.0/(beta*k))
+        r = sigma_r/sigma_r.unit*np.random.random()*units.angstrom + r0
         return r
 
     def _propose_angle(self, angle, beta):
         """
         Bond angle proposal
         """
-        theta0 = angle.type.theteq*units.radians
-        k_eq = angle.type.k*units.kilocalories_per_mole/(units.radians**2)
-        sigma = beta*2.0/np.sqrt(2.0*k_eq/k_eq.unit)
-        theta = sigma/sigma.unit*np.random.random()*theta0.unit + theta0
+        theta0 = angle.type.theteq
+        k = angle.type.k
+        sigma_theta = units.sqrt(1.0/(beta*k))
+        theta = sigma_theta/sigma_theta.unit*np.random.random()*units.radian + theta0
         return theta
 
-    def _torsion_potential(self, torsion, phi, beta):
+    def _torsion_logq(self, torsion, phi, beta):
         """
         Calculate the log-unnormalized probability
         of the torsion
         """
-        phi = phi/phi.unit
-        beta = beta/beta.unit
-        gamma = torsion.type.phase
-        V = torsion.type.phi_k
+        phi = phi
+        beta = beta
+        gamma = torsion.type.phase*units.degree
+        V = torsion.type.phi_k*units.kilocalorie_per_mole
```

Comment:
You shouldn't be assigning units here.  Can you convert these to appropriate units in `_add_torsion_units()`?